### ISSUE TITLE
Cursor doesn't properly disappear with Pipewire capture

### ISSUE THREAD

**Issue by rmnvgr** [obsproject/obs-studio#4895]
### Operating System Info

Other

### Other OS

Fedora 34

### OBS Studio Version

27.0.1

### OBS Studio Log URL

https://obsproject.com/logs/AUrBGETVVGn7lrbQ

### Expected Behavior

When the cursor disappears in the captured window/screen, it also disappears in the output.

### Current Behavior

The cursor is still visible in the output.

### Steps to Reproduce

1. Open an application where the cursor disappears (for example OpenTTD or Minecraft)
2. Add a Pipewire capture (either window or screen)
3. See that cursor still appears in the output

### Anything else we should know?

In OpenTTD (cursor in the bottom right):
![openttd](https://user-images.githubusercontent.com/66537460/121820380-ba6e1480-cc92-11eb-8926-26310b6fa886.png)

In Minecraft (cursor at the top left of the pointer):
![minecraft](https://user-images.githubusercontent.com/66537460/121820389-c35ee600-cc92-11eb-9edc-4eea80656d70.png)

**Comment by RytoEX** [comment]
Just to confirm, did you uncheck "Show Cursor" on the source properties in OBS?

![Screenshot from 2021-06-13 19-06-42](https://user-images.githubusercontent.com/624931/121824575-1cbb1b00-cc7b-11eb-8569-e2c47ebaf2da.png)

**Comment by rmnvgr** [comment]
No, I did not uncheck "Show Cursor". That would do for OpenTTD, which provides its own cursor, but not for Minecraft, which has to display the cursor in menus/inventory.

**Comment by GeorgesStavracas** [comment]
I've been trying to track this down for months now. It is almost certainly a Mutter issue. For some reason, it keeps sending cursor data even when the cursor should be invisible. It seems not even https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1417 helped this case...

**Comment by RytoEX** [comment]
Is Minecraft perhaps switching from a custom cursor to the system cursor for things like the inventory screen?  If so, I'm not sure how that would be handled, at least on our end.

**Comment by rmnvgr** [comment]
@GeorgesStavracas Is there a Mutter issue I can subscribe to already? Or do you want me to open one?

> Is Minecraft perhaps switching from a custom cursor to the system cursor for things like the menus and inventory screen? If so, I'm not sure how that would be handled, at least on our end.

Minecraft hides the cursor in-game and shows the system cursor in menus and inventory. OpenRCT2 and Julius use custom cursors, and the system cursor doesn't appear in the output.

**Comment by GeorgesStavracas** [comment]
PipeWire captures ask the desktop portal / compositor for metadata cursors. Compositors then send the monitor / window contents as the actual data of the stream, and the cursor is sent as metadata (position + size + texture) of each buffer of the stream. Games usually hide the cursor by setting an empty texture, and draw their custom cursors as part of the game itself.

In this case, what's happening is that the compositor keeps sending metadata cursors to OBS Studio, even when the game hides the cursor.

**Comment by GeorgesStavracas** [comment]
> @GeorgesStavracas Is there a Mutter issue I can subscribe to already? Or do you want me to open one?

I suspect there already is one, but I couldn't find it. It would be fantastic if you could either report it at https://gitlab.gnome.org/GNOME/mutter/-/issues - thanks!

**Comment by rmnvgr** [comment]
I opened https://gitlab.gnome.org/GNOME/mutter/-/issues/1852

**Comment by DonaldDuckEde** [comment]
Hi, i got the same. i fixed it and maked a video on it, please watch it.
https://youtu.be/UDZ6hPgEOwQ

**Comment by DanielGaaA** [comment]
Is this really compositor issue? I am on KDE 5.26.4 and I am having same issue with OBS. When screen recording with OBS I can see SW or HW cursor when hidden.
When i was testing screen sharing with firefox(https://mozilla.github.io/webrtc-landing/gum_test.html) I didn't see the red SW cursor

![image](https://user-images.githubusercontent.com/95187493/210664599-01c6563c-be17-4c13-9aae-420848139285.png)

**Comment by kkartaltepe** [comment]
> When screen recording with OBS I can see SW or HW cursor when hidden.

Chrome hides cursors even when they are visible on screen. I can only presume they just hide the cursor after it hasnt moved for so many seconds. This is just wrong in a different way.

**Comment by rmader** [comment]
Opened https://github.com/obsproject/obs-studio/pull/8467
